src/onnx/parse_matmul.cpp

```diff
@@ -106,6 +108,62 @@ struct parse_matmul : op_parser<parse_matmul>
         return all_zeros;
     }
 
+    static instruction_ref set_scale_arg(const onnx_parser::node_info& info,
+                                         const std::vector<instruction_ref>& args,
+                                         const int index)
+    {
+        instruction_ref scale_arg                            = args[index];
+        std::set<migraphx::shape::type_t> supported_dq_types = {migraphx::shape::float_type,
+                                                                migraphx::shape::half_type};
+
+        if(not(contains(supported_dq_types, scale_arg->get_shape().type())))
+        {
+            MIGRAPHX_THROW("PARSE_QUANT_DOT_SCALDED: Scales must be float or half_type");
+        }
+
+        if(scale_arg->get_shape().scalar())
+        {
+            scale_arg = info.add_instruction(make_op("unsqueeze", {{"axes", {-1}}}), scale_arg);
+        }
+
+        return scale_arg;
+    }
+
+    static instruction_ref set_scale_bias(const std::vector<instruction_ref>& args,
+                                          const int index,
+                                          const migraphx::shape& scale_arg_shape,
+                                          const instruction_ref& compare_arg,
+                                          bool& has_valid_scale_bias)
+    {
+        has_valid_scale_bias = false;
+
+        if(args.size() > index)
```

Comment:
I don't see an index defined in MatMulIntegerToFloat. Is this for some other operator. Thanks.

Reply:
Its argument index. We're doing the check here so its done for every arg

Reply:
The parser changes here due to how we handle the input arg. Arguments are shifted for the Microsoft variant